fedora_messaging/tests/unit/test_api.py

```diff
@@ -0,0 +1,165 @@
+# coding: utf-8
+
+# This file is part of fedora_messaging.
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""Tests for the :module:`fedora_messaging.api` module."""
+
+import unittest
+import mock
+
+from fedora_messaging import api
+from fedora_messaging.exceptions import PublishException
+
+
+class ConsumeTests(unittest.TestCase):
+    @mock.patch("fedora_messaging._session.ConsumerSession")
+    def test_bindings_are_dict(self, mock_session):
+        """Assert consume is working(bindings type is dict)"""
+        mock_session.return_value = mock_session
+        api.consume("test_callback", dict())
+        mock_session.assert_called_once()
+        mock_session.consume.assert_called_once_with("test_callback", [dict()])
+
+    @mock.patch("fedora_messaging._session.ConsumerSession")
+    def test_bindings_not_dict(self, mock_session):
+        """Assert consume is working(bindings type is not dict)"""
+        mock_session.return_value = mock_session
+        api.consume("test_callback", "test_bindings")
+        mock_session.assert_called_once()
+        mock_session.consume.assert_called_once_with("test_callback", "test_bindings")
+
+
+class PublishTests(unittest.TestCase):
+    @mock.patch("fedora_messaging.api._session_cache")
+    @mock.patch("fedora_messaging.api.pre_publish_signal")
+    @mock.patch("fedora_messaging.api.publish_failed_signal")
+    @mock.patch("fedora_messaging.api.publish_signal")
+    @mock.patch("fedora_messaging._session.PublisherSession")
+    def test_publish_to_exchange(
+        self,
+        mock_publisher_session,
+        mock_publish_signal,
+        mock_publish_failed_signal,
+        mock_pre_publish_signal,
```

Comment:
I think you can not mock out the signals and instead write a signal handler which calls self.fail() for ones that shouldn't be called. My general feeling is mocking should be for things that can't easily work in the test environment (e.g. the broker connection), but the signals should be simple to use as part of your tests.